File: .config/hiss/testdata/HISS-15/typescript/gap/positive-only-test.ts
// Only the positive dimension. The negative dimension (NaN, an inverted range) and the
// boundary dimension (value exactly at low and at high, Number.MAX_SAFE_INTEGER) are absent.
// node --test reports a passing suite and nothing measures the missing dimensions.
import { test } from "node:test";
import assert from "node:assert";
import { clamp } from "./untested-exported-function";

test("clamp returns the value inside the range", () => {
  assert.strictEqual(clamp(5, 0, 10), 5);
});

File: .config/hiss/testdata/HISS-15/python/gap/positive-only-test.py
"""Only the positive dimension is asserted. The negative dimension (a non-numeric argument)
and the boundary dimension (value exactly at low and at high) are absent, which HISS-15
forbids. Nothing in the repository measures test dimensions for Python."""

import unittest

from untested_public_function import clamp


class ClampTest(unittest.TestCase):
    def test_inside_range(self):
        self.assertEqual(clamp(5, 0, 10), 5)
